<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>868</width>
    <height>734</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>qmtunnel GUI</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QTreeView::branch:has-siblings:!adjoins-item {
    border-image: url(/images/stylesheet-vline.png) 0;
}

QTreeView::branch:has-siblings:adjoins-item {
    border-image: url(:/images/stylesheet-branch-more.png) 0;
}

QTreeView::branch:!has-children:!has-siblings:adjoins-item {
    border-image: url(:/images/stylesheet-branch-end.png) 0;
}

QTreeView::branch:has-children:!has-siblings:closed,
QTreeView::branch:closed:has-children:has-siblings {
        border-image: none;
        image: url(:/images/stylesheet-branch-closed.png);
}

QTreeView::branch:open:has-children:!has-siblings,
QTreeView::branch:open:has-children:has-siblings  {
        border-image: none;
        image: url(:/images/stylesheet-branch-open.png);
}</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QTreeWidget" name="browserTree">
       <property name="contextMenuPolicy">
        <enum>Qt::CustomContextMenu</enum>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Sunken</enum>
       </property>
       <property name="lineWidth">
        <number>3</number>
       </property>
       <property name="midLineWidth">
        <number>3</number>
       </property>
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <property name="dragDropMode">
        <enum>QAbstractItemView::InternalMove</enum>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="autoExpandDelay">
        <number>0</number>
       </property>
       <property name="rootIsDecorated">
        <bool>true</bool>
       </property>
       <property name="uniformRowHeights">
        <bool>true</bool>
       </property>
       <property name="sortingEnabled">
        <bool>true</bool>
       </property>
       <property name="columnCount">
        <number>2</number>
       </property>
       <attribute name="headerVisible">
        <bool>true</bool>
       </attribute>
       <attribute name="headerCascadingSectionResizes">
        <bool>true</bool>
       </attribute>
       <attribute name="headerStretchLastSection">
        <bool>false</bool>
       </attribute>
       <column>
        <property name="text">
         <string>Name</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Status &amp; Latency</string>
        </property>
       </column>
      </widget>
      <widget class="QStackedWidget" name="stackedWidget">
       <widget class="QWidget" name="page_MgrConn">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QTabWidget" name="mgrconn_tabWidget">
           <property name="currentIndex">
            <number>0</number>
           </property>
           <widget class="QWidget" name="mgrconn_tabConnection">
            <attribute name="title">
             <string>Manager connection</string>
            </attribute>
            <layout class="QVBoxLayout" name="vLayoutTabConnection" stretch="0,0">
             <item>
              <widget class="QLabel" name="lblMgrConnIntro">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>The information below is only used by GUI to connect to qmtunnel host management interface.</string>
               </property>
               <property name="wordWrap">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QWidget" name="widget_3" native="true">
               <layout class="QHBoxLayout" name="horizontalLayout_6">
                <property name="leftMargin">
                 <number>6</number>
                </property>
                <property name="topMargin">
                 <number>6</number>
                </property>
                <property name="rightMargin">
                 <number>6</number>
                </property>
                <property name="bottomMargin">
                 <number>6</number>
                </property>
                <item>
                 <widget class="QPushButton" name="btnApplyAndSave">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>Apply &amp;&amp; Save profile</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="btnApply">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>Apply</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer_4">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QPushButton" name="btnCancel">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>Cancel</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="widget_mgrServerButtons" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <property name="leftMargin">
             <number>5</number>
            </property>
            <property name="topMargin">
             <number>5</number>
            </property>
            <property name="rightMargin">
             <number>5</number>
            </property>
            <property name="bottomMargin">
             <number>5</number>
            </property>
            <item>
             <widget class="QPushButton" name="btnMgrServerConfigApply">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Apply &amp;&amp; Save</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_7">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="btnMgrServerConfigCancel">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Cancel &amp;&amp; Restore</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="page_tunnel">
        <layout class="QVBoxLayout" name="vLayoutPageTunnel">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QTabWidget" name="tunnel_tabWidget">
           <property name="currentIndex">
            <number>0</number>
           </property>
           <widget class="QWidget" name="tab_tunnelStats">
            <attribute name="title">
             <string>Statistics summary</string>
            </attribute>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <widget class="QCheckBox" name="chkTunnelStatsAutoUpdate">
                 <property name="text">
                  <string>auto-update every</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDoubleSpinBox" name="spinTunnelStatsAutoUpdateInterval">
                 <property name="suffix">
                  <string> s</string>
                 </property>
                 <property name="decimals">
                  <number>1</number>
                 </property>
                 <property name="minimum">
                  <double>0.100000000000000</double>
                 </property>
                 <property name="maximum">
                  <double>999.000000000000000</double>
                 </property>
                 <property name="value">
                  <double>2.000000000000000</double>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="horizontalSpacer">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QTreeWidget" name="tunnelStatsTreeWidget">
               <property name="editTriggers">
                <set>QAbstractItemView::NoEditTriggers</set>
               </property>
               <property name="alternatingRowColors">
                <bool>true</bool>
               </property>
               <property name="selectionMode">
                <enum>QAbstractItemView::SingleSelection</enum>
               </property>
               <property name="selectionBehavior">
                <enum>QAbstractItemView::SelectRows</enum>
               </property>
               <property name="wordWrap">
                <bool>true</bool>
               </property>
               <property name="headerHidden">
                <bool>true</bool>
               </property>
               <attribute name="headerStretchLastSection">
                <bool>false</bool>
               </attribute>
               <column>
                <property name="text">
                 <string>Parameter</string>
                </property>
               </column>
               <column>
                <property name="text">
                 <string>Value</string>
                </property>
               </column>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Last tunnel error received:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="lblTunnelLastErrorStr">
               <property name="frameShape">
                <enum>QFrame::Panel</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Sunken</enum>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="wordWrap">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="tab_tunnelConnList">
            <attribute name="title">
             <string>Connection list</string>
            </attribute>
            <layout class="QVBoxLayout" name="vLayout_tunnelConnList"/>
           </widget>
           <widget class="QWidget" name="tab_tunnelConfig">
            <attribute name="title">
             <string>Configuration</string>
            </attribute>
            <layout class="QVBoxLayout" name="vLayoutTunnelConfig">
             <item>
              <spacer name="verticalSpacer">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>513</width>
                 <height>592</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QWidget" name="widget_4" native="true">
               <layout class="QHBoxLayout" name="horizontalLayout_8">
                <property name="leftMargin">
                 <number>6</number>
                </property>
                <property name="topMargin">
                 <number>6</number>
                </property>
                <property name="rightMargin">
                 <number>6</number>
                </property>
                <property name="bottomMargin">
                 <number>6</number>
                </property>
                <item>
                 <widget class="QPushButton" name="btnTunnelConfigApply">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>Apply</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer_5">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QPushButton" name="btnTunnelConfigCancel">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>Cancel</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="page_empty"/>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>868</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;Profile</string>
    </property>
    <addaction name="action_profile_new"/>
    <addaction name="action_profile_open"/>
    <addaction name="action_profile_save"/>
    <addaction name="action_profile_saveAs"/>
    <addaction name="separator"/>
    <addaction name="action_UISettings"/>
    <addaction name="action_exit"/>
   </widget>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>&amp;?</string>
    </property>
    <addaction name="action_About"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu"/>
  </widget>
  <action name="action_profile_new">
   <property name="text">
    <string>&amp;New profile..</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="action_profile_open">
   <property name="text">
    <string>&amp;Open profile...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_profile_saveAs">
   <property name="text">
    <string>Save profile &amp;As...</string>
   </property>
  </action>
  <action name="action_exit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
  </action>
  <action name="action_profile_save">
   <property name="text">
    <string>&amp;Save profile</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_autoopen_recent_profile">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Auto-open most recent profile</string>
   </property>
  </action>
  <action name="action_UISettings">
   <property name="text">
    <string>UI Settings...</string>
   </property>
  </action>
  <action name="action_About">
   <property name="text">
    <string>&amp;About..</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
